frontend/src/pages/dashboard.jsx

```diff
@@ -0,0 +1,339 @@
+        setOpenAddDialogue(false);
+        setCategory("");
+        setAmount(0);
+        setDescription("");
+        setExpenses(response);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    createExpense();
+  };
+  return (
+    <ThemeProvider theme={theme}>
+      <TableContainer component={Paper}>
+        <Table sx={{ minWidth: 500 }} aria-label="simple table">
+          <TableHead>
+            <TableRow>
+              <TableCell>Category</TableCell>
+              <TableCell align="right">Amount</TableCell>
+              <TableCell align="right">Description</TableCell>
+              <TableCell align="left">Actions</TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {expenses.map((row) => (
+              <TableRow
+                key={row._id}
+                sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
+              >
+                <TableCell component="th" scope="row">
+                  {editRowId === row._id ? (
+                    <>
+                      <InputLabel id="category">Category</InputLabel>
+                      <Select
+                        required
+                        labelId="catergory"
+                        id="category"
+                        value={category}
+                        label="category"
+                        onChange={handleCategory}
+                      >
+                        <MenuItem value={"Wants"}>Wants</MenuItem>
+                        <MenuItem value={"Needs"}>Needs</MenuItem>
+                        <MenuItem value={"Savings"}>Savings</MenuItem>
+                      </Select>
+                    </>
+                  ) : (
+                    row.category
+                  )}
+                </TableCell>
+                <TableCell align="right">
+                  {editRowId === row._id ? (
+                    <TextField
+                      required
+                      id="amount"
+                      label="Amount"
+                      variant="outlined"
+                      type="number"
+                      value={amount}
+                      onChange={handleAmount}
+                    />
+                  ) : (
+                    row.amount
+                  )}
+                </TableCell>
+                <TableCell align="right">
+                  {editRowId === row._id ? (
+                    <TextField
+                      required
+                      id="description"
+                      label="Description"
+                      variant="outlined"
+                      type="text"
+                      value={description}
+                      onChange={handleDescription}
+                    />
+                  ) : (
+                    row.description
+                  )}
+                </TableCell>
+                <TableCell>
+                  {editRowId === row._id ? (
+                    <Button
```

Comment:
The save button is not working for me to update the line item. I get a 400 error from the API /expenses/id

Reply:
Fixed this as well.